### PR TITLE
Video-54-Add-Category-Sidebar-And-Filter

### PR DESCRIPTION
**Pull request by basir**

### CODE REVIEW

File: frontend/src/reducers/productReducers.js [hunk 1]
```diff
@@ -35,6 +38,22 @@ export const productListReducer = (
   }
 };
 
+export const productCategoryListReducer = (
+  state = { loading: true, products: [] },
```
Comment (Macronius) [inline frontend/src/reducers/productReducers.js:42]: Why on this line was 'products' not switched to 'categories' ?

Or is it simply not being utilized at all and has nothing to do with the SUCCESS categories return, and therefore could have been deleted?

File: frontend/src/actions/productActions.js [hunk 2]
```diff
@@ -15,24 +15,41 @@ import {
...
+  PRODUCT_CATEGORY_LIST_FAIL,
 } from '../constants/productConstants';
 
-export const listProducts = ({ seller = '', name = '' }) => async (
-  dispatch
-) => {
+export const listProducts = ({
+  seller = '',
+  name = '',
+  category = '',
```
Comment (luxi101) [inline frontend/src/actions/productActions.js:26]: I'm getting a : TypeError: Cannot read property 'category' of undefined

Comment (SOUGATA-DAS-1998) [reply frontend/src/actions/productActions.js:26]: I am also facing the same error

File: frontend/src/reducers/productReducers.js (hunk 1 above)
Comment (marothi-codes) [reply frontend/src/reducers/productReducers.js:42]: Most likely due to the fact that we're returning distinct products based on the category property of Product. On the models there is no dedicated Category model therefore no category._id hence products cannot be switched to categories.

**Pull request merged**